.phpcs.xml.dist

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0"?>
+<ruleset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="vendor/squizlabs/php_codesniffer/phpcs.xsd">
+
+    <file>app</file>
+    <file>bin</file>
+    <file>dev</file>
+    <file>lib</file>
+    <file>phpserver</file>
+    <file>pub</file>
+    <file>setup</file>
+
+    <exclude-pattern>pub/media/*</exclude-pattern>
+    <exclude-pattern>pub/opt/*</exclude-pattern>
+    <exclude-pattern>pub/static/*</exclude-pattern>
+
+    <arg name="basepath" value="."/>
+    <arg name="colors"/>
+    <arg value="sp"/>
+
+    <rule ref="Magento2Framework" />
```

Comment:
Looks like we should use a different rule set:
https://github.com/magento/magento2/blob/162c37401d775aa6412ea899728a48488ff470c3/dev/tests/static/framework/Magento/ruleset.xml#L8C16-L14

Reply:
Thanks for the link. I wonder if this pull request should remove that file.

What change would you propose making here? I don't fully understand what you are referring to.

If you're suggesting that we need to use a relative path to this coding standard: please note that this project requires `dealerdirect/phpcodesniffer-composer-installer`, which makes coding standards which are installed via Composer available by name. See also https://github.com/magento/magento-coding-standard/pull/436.

If you are suggesting that we should include `Magento2` here as well as `Magento2Framework`: please note that the `Magento2Framework` standard includes the `Magento2` standard since https://github.com/magento/magento-coding-standard/pull/417.

Reply:
> If you are suggesting that we should include Magento2 here as well as Magento2Framework: please note that the Magento2Framework standard includes the Magento2 standard since https://github.com/magento/magento-coding-standard/pull/417.

Ah, you're right. so looks fine.
BTW, Is it actually correct that framework depends on magneto2? Shouldn't it be opposite?

Reply:
> Ah, you're right. so looks fine. BTW, Is it actually correct that framework depends on magneto2? Shouldn't it be opposite?

I think it's the right way around. I see (the intent of) the `Magento2` coding standard to be relevant for all Magento code including third party / external modules. Whereas the `Magento2Framework` includes sniffs for things like Adobe copyright headers which are not relevant for third party code.

> Also, looks like Magento static test is using a "Magento" rule name, so looks like removing that file will break it. Or no? Also, https://github.com/magento/magento2/blob/2.4-develop/dev/tests/static/testsuite/Magento/Test/Php/LiveCodeTest.php#L239

I expected something like that, but the automated tests that are run here on GitHub don't seem to be complaining about the file that's been removed here. I suspect that it's a legacy file which wasn't cleaned up when it was taken out of use; I've not verified this idea.